### PR TITLE
Restored broken API links

### LINKED ISSUE #13505: Some gallery references are dead
These seem to be reference guide links, and can show up alongside other links that work just fine. A few examples from 3.3.0 docs:

## `Figure.scatter` link broken

Screenshot 2023-11-01 at 22 18 47

## `models.LabelSet` link broken

Screenshot 2023-11-01 at 22 18 55

### PR DESCRIPTION
**Pull request by Azaya89**
- [x] issues: fixes #13505 as well as some other minor clarifications in the set-up guide.
- [x] tests added / passed

### CODE REVIEW

**Comment by bryevdv** [comment]
I'm not opposed to making these immediate changes to remove "annotation" if that gets things working, but AFAIK we still don't have any root cause for the issue, or know why some models work with "long" paths and others only work with "short" paths.

**Comment by Azaya89** [comment]
> I'm not opposed to making these immediate changes to remove "annotation" if that gets things working, but AFAIK we still don't have any root cause for the issue, or know why some models work with "long" paths and others only work with "short" paths.

tbh, I don't know why they don't work anymore either, but I noticed it was mostly the APIs with 'annotation' in the path that were broken.

File: examples/interaction/legends/legend_hide.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
-''' The legend_hide feature is use to hide corresponding lines on a plot.
+''' The legend_hide feature is used to hide corresponding lines on a plot.
 This examples demonstrates an interactive line chart for stock prices over
 time which allows the user to click on the legend entries to hide or show
 the corresponding lines.
 
 .. bokeh-example-metadata::
     :sampledata: stocks
-    :apis: bokeh.plotting.figure.line, bokeh.palettes.Spectral4
+    :apis: bokeh.plotting.figure.line, bokeh.palettes.brewer-palettes
```
Comment (tcmetzger) [inline examples/interaction/legends/legend_hide.py]: This link still doesn't seem to show up correctly (at least in my local build) 
image

**Review by tcmetzger — CHANGES_REQUESTED** [review]
Looks great, thank you! 
I believe there is only one link that isn't working yet (see comment)!

**Comment by tcmetzger** [comment]
While reviewing this PR, I noticed https://github.com/bokeh/bokeh/issues/13529

File: examples/interaction/legends/legend_hide.py (hunk 1 above)
Comment (Azaya89) [reply examples/interaction/legends/legend_hide.py]: I have shortened it now to refer to the top level `bokeh.palettes` path.

**Review by bryevdv — APPROVED** [review]

**Review by tcmetzger — APPROVED** [review]
Thank you!

**Comment by bryevdv** [comment]
If we don't already have one, an issue or discussion about making these example link specifications consistent would be good.

**Pull request merged**